dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/webmodules/WebModule.java

```diff
@@ -74,17 +73,23 @@ public WebModule(String name, String namespace, String defaultAction) {
   }
 
   public static WebModule getModule(App app) {
+    String basePath = app.getBasePath();
+    if (app.isBundled()) {
+      basePath = "/api/apps/" + app.getKey();
+    }
+
     boolean hasIcon = app.getIcons() != null && app.getIcons().getIcon48() != null;
 
     String defaultAction = app.getLaunchUrl();
 
-    String icon = hasIcon ? app.getBaseUrl() + "/" + app.getIcons().getIcon48() : null;
+    String icon =
+        hasIcon ? basePath.replace("/api/", "/") + "/" + app.getIcons().getIcon48() : null;
 
     String description = subString(app.getDescription(), 0, 80);
 
-    String key = app.isBundled() ? AppManager.BUNDLED_APP_PREFIX + app.getKey() : app.getKey();
+    String key = app.getKey();
 
-    WebModule module = new WebModule(key, app.getBasePath(), defaultAction);
+    WebModule module = new WebModule(key, basePath, defaultAction);
```

Comment:
I am not sure how the change in basePath vs getBasePath will affect the FE; Is there any need to change this?

Reply:
The WebModule changes are now reverted.